data/mercator.py

```diff
@@ -1,60 +1,64 @@
-import math
-import re
-import warnings
+from typing import Union
 
-import cftime
 import numpy as np
 import pyresample
-import pytz
-from cachetools import TTLCache
-from netCDF4 import Dataset
 from pint import UnitRegistry
 
+import data.geo as geo
 from data.calculated import CalculatedData
+from data.model import Model
 from data.nearest_grid_point import find_nearest_grid_point
-from data.variable import Variable
-from data.variable_list import VariableList
+from data.netcdf_data import NetCDFData
 from utils.errors import APIError
 
 
-class Mercator(CalculatedData):
+class Mercator(Model):
     __depths = None
 
-    def __init__(self, url, **kwargs):
+    def __init__(self, nc_data: Union[CalculatedData, NetCDFData]) -> None:
+        super().__init__(nc_data)
         self.latvar = None
         self.lonvar = None
-
-        super(Mercator, self).__init__(url, **kwargs)
+        self.nc_data = nc_data
+        self._dataset = nc_data.dataset
```

Comment:
I think down the road we should dump `self._dataset` and simply change all references to `self.nc_data.dataset`. I think it'd help cut down on confusion. For now though this is ok.

Reply:
100% agree! I was exploring that last weekend and found that it ripples out into a lot of files beyond those touched by this PR, so I decided to wait and do it as a separate task.